views/src/main/java/io/micronaut/views/thymeleaf/EngineContextFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micronaut.views.thymeleaf;
+
+import org.thymeleaf.IEngineConfiguration;
+import org.thymeleaf.context.EngineContext;
+import org.thymeleaf.context.IContext;
+import org.thymeleaf.context.IEngineContext;
+import org.thymeleaf.context.IEngineContextFactory;
+import org.thymeleaf.engine.TemplateData;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Implementation of the {@link IEngineContextFactory} interface for {@link WebContext} support.
+ *
+ * @author Semyon Gashchenko
+ */
+class EngineContextFactory implements IEngineContextFactory {
+
+    @Override
+    public IEngineContext createEngineContext(
```

Comment:
This implementation could be made more dry

Reply:
Yeah. Thanks for comment.